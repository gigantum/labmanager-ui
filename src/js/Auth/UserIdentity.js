//vendor
import {
  graphql,
} from 'react-relay'
//mutations
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//environment
import {fetchQuery} from 'JS/createRelayEnvironment';

const userIdentityQuery = graphql`
  query UserIdentityQuery{
    userIdentity{
      id
      username
      email
      givenName
      familyName
      isSessionValid
    }
  }
`;

const UserIdentity = {
  getUserIdentity: () => {

    return new Promise((resolve, reject) =>{

      let fetchData = function(){

        fetchQuery(userIdentityQuery(), {}).then((response, error) => {

          if(response){
            resolve(response)
          }else{

            reject(response)
          }
        }).catch((error) =>{

          reject(error)
        })
      }

      fetchData()
    })
  },
  removeUserIdentity: () => {
    RemoveUserIdentityMutation(()=>{
      console.log('removed user')
    })
  }

}

export default UserIdentity
