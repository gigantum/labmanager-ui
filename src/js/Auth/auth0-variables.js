export const AUTH_CONFIG = {
  domain: 'calum.auth0.com',
  clientId: '9wFN5CZA4Z9xa1r8KZIZbevpjHG7bAab',
  callbackUrl: window.location.origin + '/callback'
}
