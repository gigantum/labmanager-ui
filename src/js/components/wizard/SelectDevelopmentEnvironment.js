//vendor
import React from 'react'
import SweetAlert from 'sweetalert-react'
import { QueryRenderer, graphql } from 'react-relay'
//components
import Loader from 'Components/shared/Loader'
//utilites
import environment from 'JS/createRelayEnvironment'
//mutations
import AddEnvironmentComponentMutation from 'Mutations/AddEnvironmentComponentMutation'

const BaseImageQuery = graphql`query SelectDevelopmentEnvironmentQuery($first: Int!, $cursor: String){
  availableDevEnvs(first: $first, after: $cursor){
    edges{
      node{
        id
        component{
          id
          repository
          namespace
          name
          version
          componentClass
        }
        author{
          id
          name
          email
          username
          organization
        }
        info{
          id
          name
          humanName
          description
          versionMajor
          versionMinor
          tags
          icon
        }
        osBaseClass
        developmentEnvironmentClass
        installCommands
        execCommands
        exposedTcpPorts
      }
      cursor
    }
    pageInfo{
      endCursor
      hasNextPage
      hasPreviousPage
      startCursor
    }
  }
}`

export default class SelectDevelopmentEnvironment extends React.Component {
  constructor(props){
  	super(props);
  	this.state = {
      'modal_visible': false,
      'name': '',
      'description': '',
      'selectedDevelopmentEnvironment': null,
      'selectedDevelopmentEnvironmentId': false,
      'show': false,
      'message': ''
    };
    this.continueSave = this.continueSave.bind(this);
  }

  /*
    click handle
    function(object): takes a development environment edge
    sets component state for selectedDevelopmentEnvironmentId and selectedDevelopmentEnvironment
  */
  _selectDevelopmentEnvironment(edge){
    this.setState({'selectedDevelopmentEnvironment': edge})
    this.setState({'selectedDevelopmentEnvironmentId': edge.node.id})
    this.props.toggleDisabledContinue(false);
  }

  /*
    function()
    gets current selectedDevelopmentEnvironment and passes variables to AddEnvironmentComponentMutation
    callback triggers and modal state is changed to  next window
  */
  continueSave(){

    let component = this.state.selectedDevelopmentEnvironment.node.component;
    this.props.toggleDisabledContinue(true);
    AddEnvironmentComponentMutation(
      this.props.labbookName,
      'default',
      component.repository,
      component.namespace,
      component.name,
      component.version,
      "clientMutationId",
      this.props.environmentId,
      this.props.connection,
      component.componentClass,
      (log) => {
        this.props.setComponent(this.props.nextWindow, this.state.name)
        this.props.buildCallback()
      }
    )
  }

  _environmentView() {
    return this.props.environmentView;
  }

  render(){

    return(
      <div className="SelectDevelopmentEnvironment">

        <p> Dev Environment</p>
        <QueryRenderer
          variables={{
            first: 20
          }}
          query={BaseImageQuery}
          environment={environment}
          render={({error, props}) =>{

              if(error){
                console.error(error)
                return(<div>{error.message}</div>)
              }else{
                if(props){
                  return(
                    <div className="SelectDevelopmentEnvironment__inner-container flex flex--column justify--space-between">
                      <div className="SelectDevelopmentEnvironment__selected-image-container">

                          {
                            (this.state.selectedDevelopmentEnvironment !== null) && (
                              <div className="SelectDevelopmentEnvironment__selected-image">
                                <img alt="" src={this.state.selectedDevelopmentEnvironment.node.info.icon} height="50" width="50" />
                                <p>{this.state.selectedDevelopmentEnvironment.node.info.humanName}</p>
                              </div>
                            )
                          }
                      </div>
                      <div className="SelectDevelopmentEnvironment__images flex flex--row flex--wrap justify--space-around">
                      {
                        props.availableDevEnvs.edges.map((edge) => {

                            return(
                              <div className={(this.state.selectedDevelopmentEnvironmentId === edge.node.id) ? 'SelectDevelopmentEnvironment__image--selected': 'SelectDevelopmentEnvironment__image'} onClick={()=> this._selectDevelopmentEnvironment(edge)} key={edge.node.id}>
                                <img alt="" src={edge.node.info.icon} height="50" width="50" />
                                <p>{edge.node.info.humanName}</p>
                              </div>
                            )
                        })
                      }


                    </div>
                    {
                      this._environmentView() && (
                        <div className="SelectBaseImage__progress-buttons flex flex--row justify--space-between">
                          <button onClick={() => this.continueSave()}>Save</button>
                        </div>
                      )
                    }
                  <SweetAlert
                      className="sa-error-container"
                      show={this.state.show}
                      type="error"
                      title="Error"
                      text={this.state.message}
                      onConfirm={() => {this.state.reject(); this.setState({ show: false, message: ''})}} />
                </div>

                )

                }else{
                  return(<Loader />)
                }
              }
          }}
        />

      </div>
      )
  }
}
