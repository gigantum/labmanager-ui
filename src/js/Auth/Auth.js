import history from 'JS/history';
import auth0 from 'auth0-js';
import { AUTH_CONFIG } from './auth0-variables';
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//store
import store from 'JS/redux/store'


export default class Auth {
  auth0 = new auth0.WebAuth({
    domain: AUTH_CONFIG.domain,
    clientID: AUTH_CONFIG.clientId,
    redirectUri: AUTH_CONFIG.callbackUrl,
    audience:  AUTH_CONFIG.audience,
    responseType: 'token id_token',
    scope: 'openid profile email user_metadata'
  });

  constructor() {
    this.login = this.login.bind(this);
    this.logout = this.logout.bind(this);
    this.handleAuthentication = this.handleAuthentication.bind(this);
    this.isAuthenticated = this.isAuthenticated.bind(this);

  }

  login() {
    this.auth0.authorize();
  }

  handleAuthentication() {

    this.auth0.parseHash((err, authResult) => {
      if (authResult && authResult.accessToken && authResult.idToken) {

        this.setSession(authResult);

      } else if (err) {
        console.error(err);
      //  alert(`Error: ${err.error}. Check the console for further details.`);
      }
    });
  }

  setSession(authResult) {

    // Set the time that the access token will expire at
    let expiresAt = JSON.stringify((authResult.expiresIn * 1000) + new Date().getTime());
    localStorage.setItem('access_token', authResult.accessToken);
    localStorage.setItem('id_token', authResult.idToken);
    localStorage.setItem('expires_at', expiresAt);
    localStorage.setItem('family_name', authResult.idTokenPayload.family_name);
    localStorage.setItem('given_name', authResult.idTokenPayload.given_name);
    localStorage.setItem('email', authResult.idTokenPayload.email);
    localStorage.setItem('username', authResult.idTokenPayload.nickname);
    //redirect to labbooks when user logs in

    let storrageRoute = sessionStorage.getItem('CALLBACK_ROUTE')
    let route = storrageRoute !== '' ? storrageRoute : `/labbooks`

    history.replace(route)
  }

  logout() {

    localStorage.removeItem('access_token');
    localStorage.removeItem('id_token');
    localStorage.removeItem('expires_at');
    localStorage.removeItem('family_name')
    localStorage.removeItem('given_name');
    localStorage.removeItem('email');
    localStorage.removeItem('username');
    RemoveUserIdentityMutation(()=>{
      //redirect to root when user logs out
      history.replace('/');
    })
  }

  isAuthenticated() {
    // Check whether the current time is past the
    // access token's expiry time
    let expiresAt = JSON.parse(localStorage.getItem('expires_at'));

    return new Date().getTime() < expiresAt;
  }
}
