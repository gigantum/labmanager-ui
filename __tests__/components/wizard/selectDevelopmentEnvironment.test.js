import React from 'react';
import SelectDevelopmentEnvironment from 'Components/wizard/SelectDevelopmentEnvironment';
import {mount} from 'enzyme'
import renderer from 'react-test-renderer';
import Auth from 'JS/Auth/Auth';

let toggleDisabledContinue = () => {}
let setComponent = () => {}
let setLabbookName = () => {}


test('Test SelectDevelopmentEnvironment rendering', () => {

  const component = renderer.create(


      <SelectDevelopmentEnvironment
        toggleDisabledContinue={toggleDisabledContinue}
        labbookName={'demo-lab-book'}
        setComponent={setComponent}
        setLabbookName={setLabbookName}
        nextWindow={'addEnvironmentPackage'}/>

  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
