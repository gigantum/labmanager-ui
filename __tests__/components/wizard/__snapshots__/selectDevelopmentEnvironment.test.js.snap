// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SelectDevelopmentEnvironment rendering 1`] = `
<div
  className="SelectDevelopmentEnvironment"
>
  <p>
     Dev Environment
  </p>
</div>
`;
